### DISCUSSION TITLE
Rate optimization setup

### DISCUSSION THREAD

**Discussion by soyroberto** [microsoft/finops-toolkit#1018] [Q&A]
**Context**: 
I've setup the Cost summary report, still need to sort how to bring more than 1 month of data.

Now I'm trying to run the Rate Optimization setup but I'm not fully clear on the data sources, as the configuration says:
https://microsoft.github.io/finops-toolkit/power-bi#-connect-to-your-data.

My configuration looks like this, see image
![Screenshot 2024-09-30 at 17 20 49](https://github.com/user-attachments/assets/025de0d2-8ab9-40de-b45a-5a78413a2bff)

And the report shows no DATA at all
![Screenshot 2024-09-30 at 17 23 49](https://github.com/user-attachments/assets/cdd62e01-34a1-489e-be61-d72353964901)

Has someone here set it up successfully? 
Thanks

**Comment by flanakin** [comment]
The "null to Table" error is generally because there's no data in storage. The report is configured to only show August data, so that may be due to not having exported August data. You can clear RangeStart and RangeEnd parameters to show all data in storage. I would recommend using Number of Months to max out the report based on your preferred maximum number of months.

Beyond this, make sure you've configured Cost Management exports and that the data is getting moved into the ingestion container in storage. The troubleshooting guide should help here: https://aka.ms/ftk/trouble

**Comment by soyroberto** [comment]
I'll be having a look, thank you. Question, is there a way to get support somehow on this? This how the ingestion container looks like:
![Screenshot 2024-10-02 at 16 48 26](https://github.com/user-attachments/assets/7e868a96-0a05-4eb6-b965-007837c2e650)

**Reply by flanakin** [reply]
We're working on official Microsoft Support. They're trying to find the right team to support it right now. 

For now, email ftk-support@ms and we can setup some time to get this resolved for you. In general, we'll just walk thru the same steps as the troubleshooting guide, so we mainly share that to help you get the quickest answer. If that's not helping, then getting on a short call is best. Email us and we'll help.

For your specific case, it looks like you've done all the right things.
✅ Exports are landing in the msexports container
✅ ADF is processing those and moving them into the ingestion container correctly
✅ Power BI is pointing to your hub instance
✅ RangeStart/End match the data you have exported

I have noticed issues with RangeStart/End in the past so the only thing I'd suggest at this point without digging into it is to clear both of those values out. If that doesn't help, email us and we can setup some time to get this resolved.

**Reply by flanakin** [reply]
@soyroberto were you able to resolve this?
